gwpy/timeseries/timeseries.py

```diff
@@ -1336,46 +1336,46 @@ def demodulate(self, f, stride=1, exp=False, deg=True):
         Demodulation is useful when trying to examine steady sinusoidal
         signals we know to be contained within data. For instance,
         we can download some data from LOSC to look at trends of the
-        amplitude and phase of Livingston's calibration line at 331.3 Hz:
+        amplitude and phase of LIGO Livingston's calibration line at 331.3 Hz:
 
         >>> from gwpy.timeseries import TimeSeries
         >>> data = TimeSeries.fetch_open_data('L1', 1131350417, 1131357617)
 
-        We can demodulate the `TimeSeries` at 331.3 Hz with a stride of once
-        per minute:
+        We can demodulate the `TimeSeries` at 331.3 Hz with a stride of one
+        minute:
 
         >>> amp, phase = data.demodulate(331.3, stride=60)
 
-        We can then plot these trends to visualize changes in the amplitude
-        and phase of the calibration line:
+        We can then plot these trends to visualize fluctuations in the
+        amplitude of the calibration line:
 
         >>> from gwpy.plot import Plot
-        >>> plot = Plot(amp, phase, separate=True, sharex=True)
+        >>> plot = Plot(amp)
+        >>> ax = plot.gca()
+        >>> ax.set_ylabel('Strain Amplitude at 331.3 Hz')
         >>> plot.show()
         """
         stridesamp = int(stride * self.sample_rate.value)
         nsteps = int(self.size // stridesamp)
-        # mix with a complex oscillator and stride through the TimeSeries,
+        # stride through the TimeSeries and mix with a local oscillator,
         # taking the average over each stride
         out = numpy.zeros(nsteps, dtype=complex).view(type(self))
```

Comment:
Can you modify this to be

```python
        out = type(self)(numpy.zeros(nsteps, dtype=complex))
```

This will result in `out` owning its own data, rather than being a view of something else:

```python
>>> TimeSeries(numpy.zeros(10, dtype=complex)).flags
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
>>> numpy.zeros(10, dtype=complex).view(TimeSeries).flags
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```

which makes modifying `out` easier downstream.

Reply:
@duncanmmacleod aha, so that's the difference between `type(self)(...)` and `(...).view(type(self))`. Thanks for the explanation, this is updated in the latest push.